picard/acousticbrainz/__init__.py

```diff
@@ -0,0 +1,221 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright 2014 Music Technology Group - Universitat Pompeu Fabra
+# Copyright 2020-2021 Gabriel Ferreira
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+from concurrent.futures import Future
+from functools import partial
+import json
+from tempfile import NamedTemporaryFile
+
+from picard.acousticbrainz.extractor import (
+    check_extractor_version,
+    precompute_extractor_sha,
+)
+from picard.config import get_config
+from picard.const import (
+    ACOUSTICBRAINZ_HOST,
+    ACOUSTICBRAINZ_PORT,
+    EXTRACTOR_NAMES,
+)
+from picard.util import (
+    find_executable,
+    load_json,
+    run_executable,
+)
+from picard.util.thread import run_task
+from picard.webservice import ratecontrol
+
+
+_acousticbrainz_extractor_sha = None
+ratecontrol.set_minimum_delay((ACOUSTICBRAINZ_HOST, ACOUSTICBRAINZ_PORT), 1000)
+
+
+def get_extractor(config=None):
+    if not config:
+        config = get_config()
+    extractor_path = config.setting["acousticbrainz_extractor"]
+    if not extractor_path:
+        extractor_path = find_extractor()
+    else:
+        extractor_path = find_executable(extractor_path)
+    return extractor_path
+
+
+def find_extractor():
+    return find_executable(*EXTRACTOR_NAMES)
+
+
+def ab_available():
+    return _acousticbrainz_extractor_sha is not None
+
+
+def ab_check_version(extractor, tagger):
+    extractor_path = find_executable(extractor)
+    return check_extractor_version(extractor_path, tagger)
+
+
+def ab_setup_extractor(tagger, config=None):
```

Comment:
`tagger` is not actually needed by this function. It is only used for `tagger.log` calls, but the usual way to use the logging is by importing `picard.log` directly:

```
from picard import log

log.debug('foo')
```

Reply:
Done.